packages/common/src/utils/challenges.ts

```diff
@@ -334,43 +334,68 @@ export const challengeRewardsConfig: Record<
 export const makeOptimisticChallengeSortComparator = (
   userChallenges: Partial<Record<ChallengeRewardID, OptimisticUserChallenge>>
 ): ((id1: ChallengeRewardID, id2: ChallengeRewardID) => number) => {
+  const playCountOrder = [
+    ChallengeName.PlayCount250,
+    ChallengeName.PlayCount1000,
+    ChallengeName.PlayCount10000
+  ]
+
+  const getSortKey = (challenge: OptimisticUserChallenge) => {
+    if (challenge.claimableAmount > 0) {
+      return 0
+    }
+    if (
+      isNewChallenge(challenge.challenge_id) &&
+      challenge.state !== 'disbursed'
+    ) {
+      return 1
+    }
+    if (challenge.state === 'disbursed') {
+      return 2
+    }
+    return 3
+  }
+
   return (id1, id2) => {
     const userChallenge1 = userChallenges[id1]
     const userChallenge2 = userChallenges[id2]
 
     if (!userChallenge1 || !userChallenge2) {
       return 0
     }
-    if (userChallenge1?.claimableAmount > 0) {
-      return -1
-    }
-    if (userChallenge2?.claimableAmount > 0) {
-      return 1
-    }
-    if (
-      userChallenge1?.challenge_id &&
-      isNewChallenge(userChallenge1?.challenge_id) &&
-      userChallenge1?.state !== 'disbursed'
-    ) {
+
+    const sortKey1 = getSortKey(userChallenge1)
+    const sortKey2 = getSortKey(userChallenge2)
+
+    if (sortKey1 < sortKey2) {
       return -1
     }
-    if (
-      userChallenge2?.challenge_id &&
-      isNewChallenge(userChallenge2?.challenge_id) &&
-      userChallenge2?.state !== 'disbursed'
-    ) {
-      return 1
-    }
-    if (userChallenge1?.state === 'disbursed') {
+    if (sortKey1 > sortKey2) {
       return 1
     }
-    if (userChallenge2?.state === 'disbursed') {
-      return -1
+
+    // If sort keys are equal, check if both are play count challenges
+    if (isPlayCountChallenge(id1) && isPlayCountChallenge(id2)) {
```

Comment:
like could we just add this clause and leave the rest as-is?

Reply:
nah doesn't sort if we just do this, we need to change the comparison a bit. can keep it more like the original code if you really want tho. 

imo the `getSortKey` function makes this all more easy to digest. we just give sort order preference in that function and the actual sort is handled inside of the comparator vs unnecessarily hand coding each comparison when they can fit in a shared form. 

lmk wyt, can change if really wanted

Reply:
fair enough, ill take it